fastlane/fastlane
# [spaceship] Initial support for updated review submissions flow

### Checklist

### Motivation and Context
Apple is introducing a [new submission workflow](https://developer.apple.com/app-store-connect/submission-update/) on App Store Connect. Once an account is updated to follow the new flow, Fastlane's current submission methods will not work. This PR contains the initial work done for adding support for new concepts like `ReviewSubmission`-s and `ReviewSubmissionItem`-s. 

While these endpoints have not been made public from Apple's side, the work done in this PR follows the observed HTTP calls that are made when creating a submission on App Store Connect.

### Description
This PR introduces the concepts of `ReviewSubmission` and `ReviewSubmissionItem`.

Support is added for:
- creating a new review submission
- fetching an existing review submission
- adding items to an existing review submission (only the App Store version, for now)
- submitting a review submission
- cancelling a review submission

The App Store Connect API supports additional concepts, states and operations which have not been included in this PR. In other words, this PR does not aim to provide an exhaustive coverage of the functionalities connected to the new submission flow; rather, its goal is to enable the development of a version of `Deliver` that is compliant with the new submission flow and that can be used when Apple eventually pulls the trigger and enforces it on all developer accounts.

Note that these new classes have not been integrated in `Deliver` yet. This was a conscious choice, as the new submission flow allows users to batch contributions of different kinds (an app version update, a new custom product page, in-app events, etc.); assuming that the submission happens instantaneously as it does now would therefore have been a very opinionated choice, which I think should be taken by the Fastlane community at large.

Rubocop is currently failing due to the increased length of the `App` class caused by the addition of the new submission methods. While I would tentatively argue that such issue could be resolved by re-issuing the rubocop file, I figured it would have been best to hear the community's opinion on the matter.
### Testing Steps


The PR includes unit tests that verify the correctness of the methods and that are executed with the usual `bundle exec rspec` command. In addition, the flow has been tested locally against my company's App Store Connect account using a slightly twisted (and opinionated) version of `Deliver`.
